### ISSUE TITLE
Vision Processing Pipeline

### ISSUE THREAD

**Issue by alecGraves** [KSU-ART/Tasks#10]
- [ ] Decide on pipeline setup, network architectures, datasets

- [ ] Neural networks process and run on image data
    - [ ] Make a detection library in tensorflow
    - [ ] make class to easily run model
    - [ ] make a ros wrapper for the library

- [ ] Collect new data
    - [ ] Set up qualification pool obstacles
    - [ ] Manual control of the sub / low-def gopro footage

**Comment by alecGraves** [comment]
@eneleski Step 2: python-based data lebeling tool, 
* draw boxes over different categories of images

**Comment by alecGraves** [comment]
data labeling tool moved to #13

**Comment by hjevans** [comment]
Github documentation for TensorFlow object detection API models, etc:
https://github.com/tensorflow/models/tree/master/research/object_detection

G-RMI powerpoint w/ info
http://presentations.cocodataset.org/Places17-GMRI.pdf

Tiny YOLO (general YOLO webpage):
https://pjreddie.com/darknet/yolo/

**Comment by alecGraves** [comment]
just an idea: use batching to more efficiently process data from multiple cameras at the same time
And maybe use a recurrent network to help the results stay continuous over time... If we do not do that, the results will be super jumpy, using tinyyolo.

Also, we should try using a custom tinyyolo architecture based on mobilenet to improve accuracy. Or a custom NAS- based model. Basically swap base models, train on mscoco, and fine-tune for our application. We will need to use tensorflow to maintain compatibility with the movidius.

**Comment by alecGraves** [comment]
TODO: Get retinanet to work, try swapping backend, etc.

**Comment by alecGraves** [comment]
We are transitioning to the movidius as the main processor for the underwater competition.

@cmcdermitt is working on making an object-detection neural network work, @eneleski  and @zcesar1 are setting up the movidius sdk and getting ready to work on movidius ros stuff.
